### ISSUE TITLE
When re-rendering component the tabs are wrapping.  

### ISSUE THREAD

**Issue by MJBNetDev** [maslianok/react-responsive-tabs#64]
I am using react-responsive-tabs to display categorized results from a search.   As the user types I am re-rendering based on the results.     There are times as the number of tabs changes the tabs wrap and do not show the "..." more tab.    However if I click on a tab that is not active it fixes itself.

```
makeTabs = (results) => results.map((item, index) => ({
        title: this.getTitle(item),
        getContent: () => this.getTabContent(item),
        /* Optional parameters */
        key: index,
        tabClassName: "category-tab",
        panelClassName: "panel"
    }))
    
    getTabs = () => {
        const tabData = this.state.groupedResults;
        return (<Tabs selectedTabKey="0" tabsWrapperClass="categorized-results" items={this.makeTabs(tabData)} transform={false}/>);
    }
```

![Screen Shot 2020-02-11 at 2 39 25 PM](https://user-images.githubusercontent.com/1383806/74272281-6046f300-4cdc-11ea-9eeb-c6c51d7d70f9.png)

Then click a tab that isn't active
![Screen Shot 2020-02-11 at 2 40 24 PM](https://user-images.githubusercontent.com/1383806/74272348-7785e080-4cdc-11ea-89ba-623759c91954.png)

Also an maybe it should be a separate issue but the selectedTabKey needs to reset to the passed in prop when the number of tabs change.

**Comment by secabhi** [comment]
I am also facing same issue. the only way around is once you render don't re-render it.
Once you setSate for tabs list don't empty the list and set again same data in list.

**Comment by rkteja** [comment]
Facing the same. I am trying to update the count so render will happen due to the tabs are getting wrapped.
If there is any work around let me know.

**Comment by maslianok** [comment]
Hey guys,

thanks for your feedback 👍 should be fixed in v4.1.0.

In addition to this now you can use an optional `uid` prop. Change it manually when you need to re-render the component based on some external data.

Let me know if you need any further help

**Comment by Sabarish-Vasudevan** [comment]
This issue is still there in v4.1.1. Is there any workaround for this? Using the optional uid prop also does not work. Any other suggestions to fix this would be greatly appreciated.

**Comment by maslianok** [comment]
@Sabarish-Vasudevan 

The component width gets recalculated on every `<Tabs />` rerender.

**_In case you change the tab title dynamically_** (e.g. your data gets fetched inside the tab component or something like this) **_you have to trigger rerender manually._**, because there is no way for the <Tabs /> to understand that something was changed.

For this you can pass a `uid` prop as I mentioned above.

---
If it doesn't solve your issue - please provide a code snippet that reproduces the issue. Preferably it should be a link to Code Sandbox http://react.new/

Thanks!

**Comment by Sabarish-Vasudevan** [comment]
> @Sabarish-Vasudevan
> 
> The component width gets recalculated on every `<Tabs />` rerender.
> 
> **_In case you change a tab title dynamically_** (e.g. your data gets fetched inside the tab component or something like this) **_you have to trigger rerender manually._**, because there is no way for the to understand that something was changed.
> 
> For this you can pass a `uid` prop as I mentioned above.
> 
> ```js-jsx
> <Tabs items={...} uid={uid} ... />
> ```
> 
> If it doesn't solve your issue - please provide a code snippet that reproduces the issue. Preferably it should be a link to Code Sandbox http://react.new/
> 
> Thanks!

This is exactly what i am doing. I am not only changing a tab's label dynamically but i am also adding a new tab dynamically. Providing a uid does not solve this issue. Please find below the link to the codesandbox. It is not an exact replica of the code as I am not at liberty to share it, but it is what i am doing. The tabs are displayed after a API call which has the list of tabs a user has created. The tabs are wrapping whenever a user adds a new tab or changes the name of the tab.

https://codesandbox.io/s/white-mountain-c19s5?file=/src/Tab.js

**Comment by maslianok** [comment]
The code you sent works as expected for me.

In which part of the app you send the API call?  
Where do you store the list of tabs? (in the component's state?)  
Do you update the component that renders the `<Tabs ... />` after you finish the API call?

**Comment by Sabarish-Vasudevan** [comment]
> The code you sent works as expected for me.
> 
> In which part of the app you send the API call?
> Where do you store the list of tabs? (in the component's state?)
> Do you update the component that renders the `<Tabs ... />` after you finish the API call?

Hey, sorry for the delayed response.
- The API call happens during the parent component's mount cycle.
- Yes, the list of tabs is stored in the component's state.
- Yes, Once the API call is done the component is updated. It is also updated whenever a tab's name is changed / whenever a new tab is added.

**Comment by maslianok** [comment]
Unfortunately, I can't reproduce the bug by the description you provided and you can't provide a code snippet where the bug reproduces.

In this example https://codesandbox.io/s/dazzling-wing-lys8h I change tabs dynamically after 5 seconds the component mounts. And everything works as expected

**Comment by kimyunhye** [comment]
@maslianok 
 There are times as the number of tabs changes the tabs wrap and do not show the "..." more tab. 

 https://codesandbox.io/s/dazzling-wing-lys8h
 Tab.js
 ` 
class Tab extends Component {
  state = {
    tabs: presidents // do not show the "..." more tab after 5 seconds
  };
 `

I'm not sure if I'm right, I sovled 
 https://codesandbox.io/s/dazzling-wing-lys8h
 Tab.js
 ` 
class Tab extends Component {
  state = {
    tabs: presidents // do not show the "..." more tab after 5 seconds
  };
...

  getTabs = () => {
    return this.state.tabs.map((president, index) => ({
      title: president.name,
      getContent: () => president.biography,
      /* Optional parameters */
      key: index + uniqueId(), // do show the "..." more tab after 5 seconds
      tabClassName: "tab",
      panelClassName: "panel"
    }));
  };
 `

**Comment by maslianok** [comment]
@kimyunhye 
are you saying that [this](https://codesandbox.io/s/dazzling-wing-lys8h) example doesn't work for you?
It doesn't show the `...` tab after 5 seconds?

Do you see any errors/warnings in console?
What browser do you use?

**Comment by kimyunhye** [comment]
I don't speak English as my first language, so my writing may not be understood

my browser chrome 74.0.3729.169 for window10 64bit

your exmple Tab.js (full)
`
import React, { Component } from "react";
import Tabs from "react-responsive-tabs";
import "react-responsive-tabs/styles.css";
import { uniqueId } from "lodash";

const presidents = [
  { name: "George Washington", biography: "..." },
  { name: "Theodore Roosevelt", biography: "..." },
  { name: "Theodore Roosevelt2", biography: "..." },
  { name: "Theodore Roosevelt3", biography: "..." },
  { name: "Theodore Roosevelt4", biography: "..." },
  { name: "Theodore Roosevelt5", biography: "..." },
  { name: "Theodore Roosevelt6", biography: "..." },
  { name: "Theodore Roosevelt7", biography: "..." }
];

class Tab extends Component {
  state = {

    // tabs: [
   //   { name: "George Washington", biography: "..." },
   //   { name: "Theodore Roosevelt", biography: "..." }
   //  ]

    tabs: presidents // do not show the "..." more tab after 5 seconds
  };
  componentDidMount = () => {
    setTimeout(() => {
      this.setState({ tabs: presidents });
    }, 5000);
  };
  getTabs = () => {
    return this.state.tabs.map((president, index) => ({
      title: president.name,
      getContent: () => president.biography,
      /* Optional parameters */
      key: index,
      tabClassName: "tab",
      panelClassName: "panel"
    }));
  };

  render() {
    return <Tabs items={this.getTabs()} uid={uniqueId()} transform={false} />;
  }
}

export default Tab;

`

**Comment by kimyunhye** [comment]
codesandbox console 
(after 5 seconds)


Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of ResizeDetector which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://fb.me/react-strict-mode-find-node
    in div (created by ResizeDetector)
    in ResizeDetector (created by Tabs)
    in div (created by Tabs)
    in Tabs (at Tab.js:38)
    in Tab (at App.js:6)
    in App (at src/index.js:9)
    in StrictMode (at src/index.js:8)

**Comment by maslianok** [comment]
> I don't speak English as my first language, so my writing may not be understood

Don't worry, your English is good enough to discuss the problem

> Warning: findDOMNode is deprecated in StrictMode

ok, I think I know the problem. Let me work on it

**Comment by maslianok** [comment]
@kimyunhye please try v4.3.0

**Comment by thanmaic** [comment]
@maslianok I still face the same issue even on the latest code. Please help resolve it.
I am using the Tabs Removal example. But still face this issue of tabs wrapping to new line below instead of showing the '...' .

```
onRemoveTab = (key, evt) => {
    evt.stopPropagation();

    // current tabs
    const { items } = this.state;

    // find index to remove
    const indexToRemove = items.findIndex((tab) => tab.key === key);

    // create a new array without [indexToRemove] item
    const newTabs = [
      ...items.slice(0, indexToRemove),
      ...items.slice(indexToRemove + 1),
    ];

    const nextSelectedIndex = newTabs[indexToRemove]
      ? indexToRemove
      : indexToRemove - 1;

    if (!newTabs[nextSelectedIndex]) {
      alert("You can not delete the last tab!");
      return;
    }

      this.setState({
        items: newTabs,
        selectedTabKey: newTabs[nextSelectedIndex].key,
      });
  };

getTabs = () => {
    return dummyData.map((president, index) => ({
      key: index,
      title: president.name,
      getContent: () => president.biography,
      tabClassName: "tab-wrapper",
      tabClassName: "tab",
      panelClassName: "panel",
    }));
  };

<div className="itemRemoval__wrapper" style={{ marginTop: 120 }}>
        <Tabs
          items={this.state.items}
          selectedTabKey={this.state.selectedTabKey}
          allowRemove
          removeActiveOnly
          onRemove={this.onRemoveTab}
          transform={false}
          showMore={true}
          uid={uniqueId()}
        />
 </div>

```

<img width="1190" alt="Screenshot 2021-04-19 at 11 02 22 PM" src="https://user-images.githubusercontent.com/50735542/115278635-4ff19a00-a163-11eb-97a2-254b5e6f5110.png">


@maslianok What changes should be done or any workaround for this issue?

I am using react: "^17.0.1"
react-responsive-tabs: "^4.3.0"
I am testing on MacBook Chrome browser.
